tauri/src/app.rs

```diff
@@ -37,30 +38,44 @@ impl App {
 }
 
 #[derive(Default)]
-pub struct AppBuilder {
-  invoke_handler: Option<InvokeHandler>,
-  setup: Option<Setup>,
-  splashscreen_html: Option<String>
+pub struct AppBuilder<T: 'static> {
+  invoke_handler: Option<InvokeHandler<T>>,
+  setup: Option<Setup<T>>,
+  splashscreen_html: Option<String>,
+  user_data: Option<T>
 }
 
-impl AppBuilder {
+// Default builder with no user_data
+impl AppBuilder<()> {
   pub fn new() -> Self {
```

Comment:
What would be more idiomatic is to have `new` take a `user_data`, and have the default be a `Default` implementation.

Reply:
I agree that would be nicer. However, my goal was to avoid breaking all users apps, due to the public api change. If we're fine with that then I could make that change. Thoughts?

Reply:
We're in late Alpha, that's fine.

Reply:
Wouldn't worry about breaking anything. As @khionu said, have the new take a user_data and then just implement the default trait.